### PR TITLE
Add more constants from sys/wait.h

### PR DESCRIPTION
**Pull request by nrc**
And fix WIFEXITED on Linux, which seemed to be wrong.

### CODE REVIEW

File: src/unix/bsd/mod.rs [hunk 1]
```diff
@@ -324,18 +324,30 @@ f! {
         }
     }
 
-    pub fn WIFEXITED(status: ::c_int) -> bool {
-        (status & 0x7f) == 0
+    pub fn WIFSTOPPED(status: ::c_int) -> bool {
+        (status & 0o177) == 0o177
```
Comment (alexcrichton) [inline src/unix/bsd/mod.rs]: Locally on OSX for me it looks like this is

```c
(_WSTATUS(x) == _WSTOPPED && WSTOPSIG(x) != 0x13)
```

so maybe this should also check `WSTOPSIG`?

File: src/unix/notbsd/mod.rs [hunk 2]
```diff
@@ -682,17 +682,33 @@ f! {
         }
     }
 
-    pub fn WIFEXITED(status: ::c_int) -> bool {
-        (status & 0xff) == 0
+    pub fn WIFSTOPPED(status: ::c_int) -> bool {
+        (status & 0x7f) == 0x7f
```
Comment (alexcrichton) [inline src/unix/notbsd/mod.rs]: For me locally this is:

```
#define __WIFSTOPPED(status)    (((status) & 0xff) == 0x7f)
```

Maybe an extra bit on the mask of `0x7f`?

File: src/unix/notbsd/mod.rs [hunk 3]
```diff
@@ -682,17 +682,33 @@ f! {
...
     }
 
-    pub fn WEXITSTATUS(status: ::c_int) -> ::c_int {
+    pub fn WSTOPSIG(status: ::c_int) -> ::c_int {
         (status >> 8) & 0xff
     }
 
+    pub fn WIFSIGNALED(status: ::c_int) -> bool {
+        (status & 0x7f) + 1 >= 2
```
Comment (alexcrichton) [inline src/unix/notbsd/mod.rs]: My local definition of this is crazy:

```c
#define __WIFSIGNALED(status) \                     
  (((signed char) (((status) & 0x7f) + 1) >> 1) > 0)
```

I'll... just assume this is the same!

File: src/unix/notbsd/mod.rs [hunk 4]
```diff
@@ -682,17 +682,33 @@ f! {
...
+    }
+
     pub fn WTERMSIG(status: ::c_int) -> ::c_int {
         status & 0x7f
     }
+
+    pub fn WIFEXITED(status: ::c_int) -> bool {
+        (status & 0x7f) == 0
```
Comment (alexcrichton) [inline src/unix/notbsd/mod.rs]: Good catch on this one!

File: src/unix/bsd/mod.rs (hunk 1 above)
Comment (nrc) [inline src/unix/bsd/mod.rs]: But this is `WIFSTOPPED`, not `WSTATUS`

Comment (nrc) [inline src/unix/bsd/mod.rs]: For the bsd headers I see `#define WIFSTOPPED(x)	(_WSTATUS(x) == _WSTOPPED)`
Where `#define	_WSTOPPED	0177` and `#define	_WSTATUS(x)	(_W_INT(x) & 0177)`.

File: src/unix/notbsd/mod.rs (hunk 2 above)
Comment (nrc) [inline src/unix/notbsd/mod.rs]: fixed

File: src/unix/notbsd/mod.rs (hunk 3 above)
Comment (nrc) [inline src/unix/notbsd/mod.rs]: I'm pretty sure they are

**Pull request merged**